### PR TITLE
DRAFT: Limit Validation Retries

### PR DESCRIPTION
**Pull request by dcgoodwin2112**
## Description
WIP. Here is a rough draft of the Mock API code to support limiting the number of retries authenticating

This PR adds mock user data to the process, checks if `lastName` and `last4` values in the session match the values stored for the mock user, and keeps track of the number of failed login attempts.

This does not block login if the max number of attempts has been reached yet. 

## Original issue(s)
department-of-veterans-affairs/va.gov-team#29983

## Screenshots

Screen Shot 2021-10-04 at 2 42 58 PM

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/check-in/api/local-mock-api/phase.3.js [hunk 1]
```diff
@@ -9,6 +9,8 @@ const featureToggles = require('./mocks/feature.toggles');
 const delay = require('mocker-api/lib/delay');
 
 let hasBeenValidated = false;
+const retries = { current: 0, max: 3 };
+const mockUser = { last4: '5555', lastName: 'lastName' };
```
Comment (mdewey) [inline src/applications/check-in/api/local-mock-api/phase.3.js:13]: Everything looks good so far. Does this cause this combination to be the only way to get past the validate page? 

If so , can you add that to the README?

**Comment by dcgoodwin2112** [comment]
> Looks good to me. Just to double-check, does that error message line up with the API returns when the information is wrong?

It does not unfortunately. I just tested it now that I have access and it looks like it's just the generic VA900 error. We may need some back-end changes so that we can differentiate failed login from other errors.

File: src/applications/check-in/api/local-mock-api/phase.3.js (hunk 1 above)
Comment (dcgoodwin2112) [reply src/applications/check-in/api/local-mock-api/phase.3.js]: Good point - that is the case unfortunately. I can update the readme if this is the approach we end up going with.

**Comment by acrollet** [comment]
I've tried this and it works as expected in my testing for the first visit. Once I've entered fault information 3 times, `{"maxValidateLimit":true}` is set in the session. At that point, if I start over, here's the behavior I get:

- lastName and Last4 fields are displayed
- if I enter the correct information, I can see my appt list
- if I enter faulty information, I'm immediately shown the error message without more opportunities to re-try.

Is that the behavior we want on re-load?

**Pull request closed without merging**